File: source/downloads/code/LeetCode/Python/Palindrome-Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 :
            return False
        tmp = x
        num = 0
        # 确定位数
        while tmp/10 != 0:
            tmp =tmp/10
            num += 1
        # 个位数永远都是
        if num == 0:
            return True
        
        y = x
        # x从最高位，y从个位
        while y != 0 and x != 0:
            if x/(10**num) != y%10:
                return False
            x = x%(10**num)
            y = y/10
            num = num-1
        return True

File: downloads/code/LeetCode/Java/Palindrome-Number.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x<0) return false;
        int p = x;
        int q = 0;
        //　ｑ是ｘ的反转版本，除了最高位，因为怕overflow
        while (p>=10){
            q = 10*q + p%10;
            p = p/10;
        }
        return (q==x/10) && (p == x%10);
    }
}
